src/engine/runner/shared/resource.extractor.ts

```diff
@@ -0,0 +1,22 @@
+import { unlinkSync } from "fs";
+import { join } from "path";
+import * as rimraf from "rimraf";
+import { FileParser } from "./file.parser";
+
+export class ResourceExtractor {
+  constructor(private dirname: string) {}
+
+  public removeFiles(paths: string[]) {
+    paths.forEach((path) => unlinkSync(join(this.dirname, path)));
+  }
+
+  public removeDirs(paths: string[]) {
+    paths.forEach((path) => rimraf.sync(join(this.dirname, path)));
```

Comment:
Why not `fs.rmdir`?

Reply:
Because `rmdir` can remove an only empty directory. I am removing the directory with files.